ARCHITECTURE.md

```diff
@@ -47,7 +47,20 @@ The following sections will cover some part of that workflow more in depth.
 
 ## C# / Rust Interoperability
 
-To be documented.
+The interoperability between the C# app and the Rust profilers is done with autogenerated bindings thanks to [FFIDJI](https://github.com/ogxd/ffidji).
+The file `src/DrDotnet.Interop/Interop.xml` describes the Rust methods that the C# app can call, along with all its associated types.    
+Based on this definition, FFIDJI will generate `Interop.cs` and `Interop.rs`.
+
+### How to make changes to bindings?
+- If you are on Windows, the FFIDJI executable is included at `src/DrDotnet.Interop/ffidji.exe`.  
+Otherwise, you must clone https://github.com/ogxd/ffidji and build it.
+- Run `ffidji -f csharp Interop.cs -t rust interop.rs -i Interop.xml`
+- Copy `interop.rs` to `src/DrDotnet.Profilers/` (and replace).  
+Code generation will only generate types and method signatures, so the rust method implementations must be done manually.  
+The C# one is automatically updated so there is nothing to do on this side.
+
+  | WARNING: Remember that this interoperability loads the profilers library in the DrDotnet process, not in the profiling target process, so you cannot set variables in memory to use them during the profiling session |
```

Comment:
"profiler UI" is a bit misleading IMHO since this is not the profiler and not really only an UI (it has a backend), plus it's not the UI of the profiler but rather the UI of the profiling framework since there are N profilers. I think the "process" terminology is more suited as a program is an exchangeable term. We already use "process" everywhere in DrDotnet (ProcessDiscovery, System.Diagnostics.Process APIs, process namespace sharing, ...).    
We could replace "the profiling target process" with "the profiled process" that is indeed a little clearer.